code/modules/clothing/masks/hailer.dm

```diff
@@ -0,0 +1,165 @@
+			message = phrase_list[key]
+			to_chat(usr,"<span class='notice'>You set the restrictor to: [message]</span>")
+		if(2)
+			phrase = (phrase < 11 && phrase >= 7) ? (phrase + 1) : 7
+			key = phrase_list[phrase]
+			message = phrase_list[key]
+			to_chat(usr,"<span class='notice'>You set the restrictor to: [message]</span>")
+		if(3)
+			phrase = (phrase < 18 && phrase >= 12 ) ? (phrase + 1) : 12
+			key = phrase_list[phrase]
+			message = phrase_list[key]
+			to_chat(usr,"<span class='notice'>You set the restrictor to: [message]</span>")
+		if(4)
+			phrase = (phrase < 18 && phrase >= 1 ) ? (phrase + 1) : 1
+			key = phrase_list[phrase]
+			message = phrase_list[key]
+			to_chat(usr,"<span class='notice'>You set the restrictor to: [message]</span>")
+		else
+			to_chat(usr, "<span class='notice'>It's broken.</span>")
+
+/obj/item/clothing/mask/gas/sechailer/emag_act(mob/user)
+	if(safety)
+		safety = 0
+		to_chat(user, "<span class='warning'>You silently fry [src]'s vocal circuit with the cryptographic sequencer.")
+	else
+		return
+	
+/obj/item/clothing/mask/gas/sechailer/attackby(obj/item/I, mob/user)
+	if(I.is_screwdriver())
+		switch(aggressiveness)
+			if(1)
+				to_chat(user, "<span class='notice'>You set the aggressiveness restrictor to the second position.</span>")
+				aggressiveness = 2
+				phrase = 7
+			if(2)
+				to_chat(user, "<span class='notice'>You set the aggressiveness restrictor to the third position.</span>")
+				aggressiveness = 3
+				phrase = 13
+			if(3)
+				to_chat(user, "<span class='notice'>You set the aggressiveness restrictor to the fourth position.</span>")
+				aggressiveness = 4
+				phrase = 1
+			if(4)
+				to_chat(user, "<span class='notice'>You set the aggressiveness restrictor to the first position.</span>")
+				aggressiveness = 1
+				phrase = 1
+			if(5)
+				to_chat(user, "<span class='warning'>You adjust the restrictor but nothing happens, probably because its broken.</span>")
+	if(I.is_wirecutter())
+		if(aggressiveness != 5)
+			to_chat(user, "<span class='warning'>You broke it!</span>")
+			aggressiveness = 5
+	if(I.is_crowbar())
+		if(!hailer)
+			to_chat(user, "<span class='warning'>This mask has an integrated hailer, you can't remove it!</span>")
+		else
+			var/obj/N = new /obj/item/clothing/mask/gas/half(src.loc)
+			playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+			N.fingerprints = src.fingerprints
+			N.fingerprintshidden = src.fingerprintshidden
+			N.fingerprintslast = src.fingerprintslast
+			N.suit_fibers = src.suit_fibers
+			if(!isturf(N.loc))
+				user.put_in_hands(hailer)
+				user.put_in_hands(N)
+			else
+				hailer.loc = N.loc
+			qdel(src)
+			return
+	..()
+ 
+/obj/item/clothing/mask/gas/sechailer/verb/halt()
+	set name = "HALT!"
+	set category = "Objects"
+	set desc = "Activate your face mask hailer."
+	var/key = phrase_list[phrase]
+	var/message = phrase_list[key]
+
+	if(cooldown < world.time - 35) // A cooldown, to stop people being jerks
+		if(!safety)
```

Comment:
`safety` is only touched by emag_act, so message should probably just be set there.